src/spath.c

```diff
@@ -1368,6 +1374,8 @@ spath* spath_relative(const spath* src, const spath* dst)
       __FILE__, __LINE__
     );
   }
+  spath_reduce(src);
+  spath_reduce(dst);
```

Comment:
```suggestion
```
We don't want to reduce src or dst in this function.

Reply:
I have a test that fails without it:
  sp = spath_from_str("1/2/3");
  sp1 = spath_from_str("1/2/./3/4");
  sp_rel = spath_relative(sp,sp1);
  spath_strcpy(buff,100, sp_rel);
  if(strcmp(buff, "4") != 0 ){
    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
    printf("test_spath failed, spath_is_child() failed \n");
    return TEST_FAIL;

Reply:
Let's fix the test case to match what the function does.  A few options, all fine by me... As written, I think ``spath_relative`` should return a path like ``.././3/4`` for this example.  Or we could change ``1/2/./3/4`` to be ``1/2/3/4``, or we could leave that string as is, but call ``spath_reduce(sp1)`` before calling ``spath_relative``.